### ISSUE TITLE
Document configuration files needed to run the GMLC

### ISSUE THREAD

**Issue by angrygreenfrogs** [RestComm/gmlc#45]
```
11:15:30,554 WARN  [ManagementImpl] (main) Failed to load the SCTP configuration file. 
/home/eross/RestComm/jboss-5.1.0.GA/server/default/data/SCTPManagement_sctp.xml (No such file or directory)
```

The GMLC code-base itself has a file like this here:
/gmlc/core/bootstrap/src/main/config/data

Should that file be copied in by the GMLC release build? Or is it supposed to come from another source?

**Comment by angrygreenfrogs** [comment]
I'm going to try to figure out how to do this with a test system we have access to, and write up some notes about what I had to do.

**Comment by FerUy** [comment]
Great initiative @angrygreenfrogs, looking forward to see it working :+1:

**Comment by deruelle** [comment]
Similarly to USSD Gateway for HTTP Servlet RA you can reuse some of the USSD structure for GMLC as it should be pretty similar.

**Comment by angrygreenfrogs** [comment]
This issue was raised to resolve some of the configuration errors you see when first running the unconfigured GMLC.

Including:

1) SCTP reporting that "SCTPManagement_sctp.xml" is missing
2) Documenting and explaining GmlcManagement_gmlcproperties.xml, which at this time we have some default values mentioned:

```
<?xml version="1.0" encoding="UTF-8" ?>
<gmlcgt value="628184422892"/>
<gmlcssn value="145"/>
<hlrssn value="6"/>
<mscssn value="8"/>
<maxmapv value="3"/>
```

**Comment by angrygreenfrogs** [comment]
Attached is a package of config files that were taken from a running GMLC instance that gives some examples and ideas.

This includes these files:

Mtp3UserPart_m3ua.xml
SccpStack_management2.xml
SccpStack_sccpresource.xml
SccpStack_sccpresource2.xml
SccpStack_sccprouter.xml
SccpStack_sccprouter2.xml
SCTPManagement_sctp.xml

I'm not clear on the purpose of each of these (or their contents), but looks like they are configuration for some of the stack components (SCTP connection to HLR for example)

[config.zip](https://github.com/RestComm/gmlc/files/375798/config.zip)

**Comment by FerUy** [comment]
Hi @angrygreenfrogs... those config files are extremely needed for SS7/SIGTRAN connection/association establishment with the Core Network (GSM-based or more properly said, UMTS -3G- circuit-switched CN -for GSM only, legacy E1 connections would be needed, SIGTRAN appears in UMTS which is GSM "upgrade" by 3GPP). 

Without them, the GMLC becomes useless, as it would be isolated. Important SS7/SIGTRAN know-how is needed to understand each of the parameters included there and actually make them work in a production or even a testing environment (with jSS7 simulator). 

They can also be generated from the jss-management-console (GUI) or the the jss7 CLI.

**Comment by deruelle** [comment]
Let's have a config out of the box that works flawlessly with jss7 simulator to test a single location so it's easier for people to use and test the GMLC and ensure it works correctly out of the box

**Comment by FerUy** [comment]
Agreed @deruelle :+1:

**Comment by angrygreenfrogs** [comment]
@FerUy 

I'm trying to figure out what's the minimal configuration required to get this working.

I have access to an operator test system, and looking to get the GMLC talking to it.

What I've been provided with is 2 links to MSC(s).

Each link has the following properties:

GMLC GT
Point code
GMSC IP
Port
GMSC Office ID
AS ID

AS Role as GMSC: IPSP Server
Routing Context: None

I can see where some these values fit into the configuration files, but there's a lot going on in there.

Could you possibly point me in the right direction for any documentation on how to setup the config and what all of the values mean?

**Comment by FerUy** [comment]
Sure @angrygreenfrogs 
Please see the document attached and come back to me with all of those parameters values (if it's none, leave it blank or just put None). As you are using SIGTRAN, don't pay attention to the last section (conf for connection over legacy E1).
Then we'll have to do the configuration through the jSS7 CLI, I can help you with that.
[SS7_Configs.xlsx](https://github.com/RestComm/gmlc/files/378256/SS7_Configs.xlsx)

**Comment by angrygreenfrogs** [comment]
Great, thanks @FerUy .

Check out the attached. I've tried to determine what goes where, but some of the naming for the fields they provided makes it a bit tricky, would like to hear what you think.

It'd be great to learn how to do this myself as well if you can point me towards the CLI tool you're talking about?

[SS7_Configs_MB.xlsx](https://github.com/RestComm/gmlc/files/378276/SS7_Configs_MB.xlsx)

**Comment by deruelle** [comment]
Tied to Issue #52

**Comment by FerUy** [comment]
Sorry @angrygreenfrogs for the little delay (I was in a call).
I believe the best action is sending the fill-in-the-blanks file I sent you in the first place to the operator, for core network staff review. We need all of those parameters well defined.
The peer side is an STP (Signaling Transfer Point), which performs GTT (Global Title Translation).
Once you got that, I can show you how to use the CLI to perform the correspondent configuration.

**Comment by FerUy** [comment]
@angrygreenfrogs... reviewing open issues, I see this one has been followed via email too, and has two sides: one for the configuration of the MNO you are dealing with, the other for testing using SS7 simulator. For the first case, I already provided you via email the configuration commands. As it involves a configuration setup with a customer, let's continue that via that private channel.

Regarding configuration for SS7 simulator and these files: 
Mtp3UserPart_m3ua.xml
SccpStack_management2.xml
SccpStack_sccpresource.xml
SccpStack_sccpresource2.xml
SccpStack_sccprouter.xml
SccpStack_sccprouter2.xml
SCTPManagement_sctp.xml
please refer to what I commented today in #53 and #58.

So, I'm closing this issue as it's replicated and detailed explanation of this configuration files will be covered as for #52.

**Comment by FerUy** [comment]
And btw, the config out of the box that works flawlessly with jss7 simulator is fixed by 3ae49d3 and 65cb9a1.
